drivers/hmis/app/graphql/sources/active_record_association.rb

```diff
@@ -0,0 +1,17 @@
+# https://gist.github.com/itkrt2y/1e1a947c71772044f5d67f358b4772fc
+
+class Sources::ActiveRecordAssociation < ::GraphQL::Dataloader::Source
+  def initialize(association_name, scope = nil)
+    @association_name = association_name
+    @scope = scope
+  end
+
+  def fetch(records)
+    ::ActiveRecord::Associations::Preloader.new.preload(records, @association_name, @scope)
+    records.map { |record| record.public_send(@association_name) }
```

Comment:
Is this where we'd need to add in pagination?  Something like: 
```
records.map { |record| record.public_send(@association_name).order(sort_column => :asc).limit(25).offset(50) }
```

Reply:
We probably want to think a bit more in-depth about patterns around pagination. It may be something that goes here, or it may be that we need an additional data loader to handle it. Currently paginated sub-fields will cause N+1 queries, so we probably need to do some more work around how to query properly for those fields to avoid the N+1s.